app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb

```diff
@@ -19,14 +19,19 @@ def create
       update_session(permissions.school_urn, permissions)
       redirect_to school_path
     else
-      Auditor::Audit.new(nil, 'dfe-sign-in.authorisation.failure', current_session_id).log_without_association
-      Rails.logger.warn("Hiring staff not authorised: #{oid} for school: #{selected_school_urn}")
-      redirect_to page_path('user-not-authorised')
+      not_authorised
     end
   end
 
   private
 
+  def not_authorised
+    Auditor::Audit.new(nil, 'dfe-sign-in.authorisation.failure', current_session_id).log_without_association
+    Rails.logger.warn("Hiring staff not authorised: #{oid} for school: #{selected_school_urn}")
+    flash[:hidden] = identifier
```

Comment:
Can you change this to use an instance variable please. Using a hidden filter seems to require a lot of unecessary tweaks,

Reply:
I used flash because I haven't been able to get instance variables working with the high_voltage gem. I've not used this gem before so I might be missing something. Do you know how to do it? Or do I need to use a regular controller/view instead?

Reply:
Going to refactor this out to a new view without high_voltage